icu4j/main/classes/core/src/com/ibm/icu/number/Scale.java

```diff
@@ -40,7 +40,10 @@ private Scale(int magnitude, BigDecimal arbitrary) {
     private Scale(int magnitude, BigDecimal arbitrary, MathContext mc) {
         if (arbitrary != null) {
             // Attempt to convert the BigDecimal to a magnitude multiplier.
-            arbitrary = arbitrary.stripTrailingZeros();
+            arbitrary =
+                arbitrary.unscaledValue().equals(BigInteger.ZERO)
```

Comment:
Optional: Looks like it should work, but it's not normally how I would choose to determine if a BigDecimal equals zero.  There are a few other ways which may be better or faster here:

https://stackoverflow.com/questions/10950914/how-to-check-if-bigdecimal-variable-0-in-java

Reply:
Please do whatever changes you think would be useful, I just thought it could be helpful to upload the patch (which I had previously uploaded to Jira) also to GitHub. I'm not trying to take over this bug from you.